src/learner.cc

```diff
@@ -332,8 +339,15 @@ class LearnerImpl : public Learner {
       }
     }
     {
-      // Write `predictor`, `gpu_id` parameters as extra attributes
-      for (const auto& key : std::vector<std::string>{"predictor", "gpu_id"}) {
+      std::vector<std::string> saved_params{"predictor", "n_gpus", "gpu_id"};
```

Comment:
Why is `n_gpus` here.  Is it from an old commit?

Reply:
legacy code.

Reply:
@chenqin But it's not in the master branch now?

Reply:
removed.